include/mppic/motion_model.hpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 FastSense, Samsung Research
+#pragma once
+
+#include "mppic/models/state.hpp"
+#include <xtensor/xmath.hpp>
+#include <xtensor/xmasked_view.hpp>
+
+#include "mppic/parameters_handler.hpp"
+
+namespace mppi
+{
+
+enum class MotionModelType
+{
+  DiffDrive,
+  Omni,
+  Ackermann
+};
+
+
+inline bool isHolonomic(MotionModelType type)
+{
+  switch (type) {
+    case MotionModelType::Omni:
+      return true;
+    case MotionModelType::Ackermann:
+    case MotionModelType::DiffDrive:
+      return false;
+  }
+
+  return false;
+}
+
+class IMotionModel
+{
+public:
+  virtual ~IMotionModel() = default;
+
+  virtual MotionModelType getMotionModelType() = 0;
+
+  virtual void applyConstraints(models::State & /*state*/) {}
+  bool isHolonomic() {return ::mppi::isHolonomic(getMotionModelType());}
+};
+
+class AckermannMotionModel final : public IMotionModel
```

Comment:
override - so that someone could use the `AckermannMotionModel` as the base of their own customizations in the future, if need be (for projection, mostly)

Reply:
done

Reply:
I didn't mean literally `override`, you can just do `class NAME : public NAME`

Reply:
oh, why did I do it